server/src/main/java/org/elasticsearch/action/inference/InferenceAction.java

```diff
@@ -1,18 +1,22 @@
 /*
  * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
  * or more contributor license agreements. Licensed under the Elastic License
- * 2.0; you may not use this file except in compliance with the Elastic License
- * 2.0.
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
  */
 
-package org.elasticsearch.xpack.core.inference.action;
+package org.elasticsearch.action.inference;
```

Comment:
I am not sold on this idea. It seems like we need a "new" thing thats pluggable that makes remote calls.

I don't know what ML is thinking around this.

Reply:
I was planning to ask @davidkyle about this, so perfect timing 😉 

David, we need to invoke the `InferenceAction` from server. One way of doing this, that is implemented in this PR, is moving InferenceAction and related structures to `server`. You can check the changes that would be needed.

Is there any other option you think is preferable here?

@benwtrent you mention a new way of having pluggable actions. Is there anything on your mind? A SPI-like interface of sorts?

Reply:
I changed to using a new `InferenceProviderPlugin` type, that provides an `InferenceProvider` interface. This will be implemented by the existing InferencePlugin to provide an implementation.

LMKWYT!